bundles/binding/org.openhab.binding.bticino/src/main/java/org/openhab/binding/bticino/internal/BticinoDevice.java

```diff
@@ -264,30 +465,293 @@ else if (p_protocol_read.getProperty("messageType").equalsIgnoreCase("CEN Basic
 
-                    if (p_protocol_read.getProperty("messageDescription").equalsIgnoreCase("Automation UP")) {
-                        eventPublisher.postUpdate(l_item.getName(), UpDownType.UP);
-                    } else if (p_protocol_read.getProperty("messageDescription").equalsIgnoreCase("Automation DOWN")) {
-                        eventPublisher.postUpdate(l_item.getName(), UpDownType.DOWN);
+                    switch (Integer.parseInt(l_binding_config.what)) {
+                        case 0:
+                            // Status Actual Temperature
+                        case 12:
+                            // Status Active Set Point Temperature
+                        case 13:
+                            // Status Offset Set Point Temperature
+                        case 14:
+                            // Status Set Point Temperature
+                            if (p_protocol_read.getProperty("what").equalsIgnoreCase(l_binding_config.what)
+                                    && p_protocol_read.getProperty("hStatus").equalsIgnoreCase("0")
+                                    && p_protocol_read.getProperty("temperature") != null) {
+                                logger.debug("T : {} : {}", l_item.getName(),
+                                        p_protocol_read.getProperty("temperature"));
+                                eventPublisher.postUpdate(l_item.getName(),
+                                        DecimalType.valueOf(p_protocol_read.getProperty("temperature")));
+                            }
+                            break;
+
+                        case 100:
+                            // Status Operation Mode
+                            if (p_protocol_read.getProperty("hStatus").equalsIgnoreCase("1")) {
+                                logger.debug("T_ControlStatus : {}", l_item.getName());
+                                hMainCtrlProbeState = "";
+                                hMainCtrlFailure = "";
+                                eventPublisher.postUpdate(l_item.getName(),
+                                        DecimalType.valueOf(p_protocol_read.getProperty("what")));
+                            }
+                            break;
+
+                        case 101:
+                            // Status Control Mode
+                            if (p_protocol_read.getProperty("hStatus").equalsIgnoreCase("2")) {
+                                logger.debug("T_ControlMode : {}", l_item.getName());
+                                eventPublisher.postUpdate(l_item.getName(),
+                                        DecimalType.valueOf(p_protocol_read.getProperty("what")));
+                            }
+                            break;
+
+                        case 102:
+                            // Status Main Unit Remote
+                            if (p_protocol_read.getProperty("hStatus").equalsIgnoreCase("3")) {
```

Comment:
No point in using `equalsIgnoreCase()` here

Reply:
OK
see commit https://github.com/openhab/openhab1-addons/pull/5486/commits/c1c8f3a86ba8641c7333467de21a0ba489322073